03 - CSS Variables/index-START.html

```diff
@@ -1,51 +1,111 @@
-    <input id="blur" type="range" name="blur" min="0" max="25" value="10" data-sizing="px">
+      <label for="blur">Blur:</label>
+      <input
+        id="blur"
+        type="range"
+        name="blur"
+        min="0"
+        max="25"
+        value="10"
+        data-sizing="px"
+      />
 
-    <label for="base">Base Color</label>
-    <input id="base" type="color" name="base" value="#ffc600">
-  </div>
+      <label for="base">Base Color</label>
+      <input id="base" type="color" name="base" value="#ffc600" />
+    </div>
 
-  <img src="https://source.unsplash.com/7bwQXzbF6KE/800x500">
+    <img src="https://source.unsplash.com/7bwQXzbF6KE/800x500" />
 
-  <style>
+    <style>
+      /* unless specific cases, define CSS variables on the highest component possible to make them accessible to all elements */
 
-    /*
+      /* define default variables needed in root */
+      :root {
+        --base: rgb(182, 142, 142);
+        --spacing: 1rem;
+        --blur: 1rem;
+      }
+
+      /*
       misc styles, nothing to do with CSS variables
-    */
-
-    body {
-      text-align: center;
-      background: #193549;
-      color: white;
-      font-family: 'helvetica neue', sans-serif;
-      font-weight: 100;
-      font-size: 50px;
-    }
-
-    .controls {
-      margin-bottom: 50px;
-    }
-
-    input {
-      width: 100px;
-    }
-  </style>
-
-  <script>
-  </script>
-
-</body>
+      */
+
+      body {
+        text-align: center;
+        background: #000;
+        color: white;
+        font-family: 'helvetica neue', sans-serif;
+        font-weight: 100;
+        font-size: 50px;
+      }
+
+      .controls {
+        margin-bottom: 50px;
+      }
+
+      input {
+        width: 100px;
+      }
+
+      .hl {
+        color: var(--base);
+      }
+
+      img {
+        background: var(--base);
+        padding: var(--spacing);
+        filter: blur(var(--blur));
+      }
+    </style>
+
+    <script>
+      //// selectors
+      const inputs = document.querySelectorAll('.controls input');
+
+      //// callbacks
+      const handleUpdate = (evt) => {
+        const currentInput = evt.target;
+
+        // use the dataset to get the unit of measure (if any)
+        const unitOfMeasure = currentInput.dataset.sizing || '';
+
+        // select the root with documentElement and setProperty based on the name of the input
+        // the first argument of setProperty will be the name of the variable
+        // the second argument will be its value
+        document.documentElement.style.setProperty(
+          `--${currentInput.name}`,
+          `${currentInput.value}${unitOfMeasure}`
+        );
+      };
+
+      //// listeners
+      // listen for both change and mousemove to make things more interactive
+      // change will react to the actual change of the input value
+      // mousemove to dragging or moving within the input
+      inputs.forEach((input) => input.addEventListener('change', handleUpdate));
```

Comment:
you are looking over the same thing twice here. and one time the code uses 1 line and the other time 3 lines, which makes it even harder to figure out that this is the same thing but for two different events. i find this is confusing.

I know, this seems very detailed here, because we are only dealing with 100 lines of code here. but soon you will be working with 100000000000000000000000000000 lines of code :)

Reply:
got it! I'm still trying to understand whether it is only Prettier playing a role here or also something else. By changing the tabWidth to 1 in the Prettier settings I got rid of the automatic indentation to 3 lines.